graph_snyk_cve_sync.py

```diff
@@ -190,124 +226,186 @@ def get_affected_versions(self, rules, versions):
                                 affected.append(ver)
         return list(set(affected))
 
-    def get_affected_versions_npm(self, rules, versions):
-        """Get affected versions for npm."""
-        affected = []
-        for ver in versions:
-            for rule in rules:
-                if Version(ver) in rule:
-                    affected.append(ver)
-                    break
-        return list(set(affected))
-
     def get_semver_versions(self, versions):
         """Convert to semver version format."""
         semver = []
         for ver in versions:
             semver.append(api.to_semver(ver))
         return semver
 
-    def extract_data_from_feed(self, data):
+    def extract_data_from_feed(self):
         """Fetch all the required info from the feed."""
-        for eco in data:
-            if eco == "java":
-                self.parse_maven_data(data[eco])
-            elif eco == "js":
-                self.parse_npm_data(data[eco])
-            elif eco == "python":
-                self.parse_pypi_data(data[eco])
+        for eco in self.snyk_data:
+            if eco == "java" and self.parse_data_for_eco(eco):
+                logger.info("Parsing feed for Maven.")
+                self.parse_maven_data(self.snyk_data[eco])
+            elif eco == "js" and self.parse_data_for_eco(eco):
+                logger.info("Parsing feed for Npm.")
+                self.parse_npm_data(self.snyk_data[eco])
+            elif eco == "python" and self.parse_data_for_eco(eco):
+                logger.info("Parsing feed for Pypi.")
+                self.parse_pypi_data(self.snyk_data[eco])
+            else:
+                logger.info("Ignoring the ecosystem {} from the feed".format(eco))
 
     def parse_maven_data(self, maven_data):
         """Parse data for maven."""
+        total_vuln = 0
```

Comment:
A lot of common code across all the parse functions, can be extracted into a common function.

Reply:
actually these count and all for each ecosystem. This is just for logging purpose. Used at the end of the function to print some stats on the pkg count, vuln count etc

Reply:
Lets look at this in the next iteration with regard to modularity